backend/src/main/java/se/chalmers/dat265/group1/api/material/MaterialsController.java

```diff
@@ -0,0 +1,103 @@
+package se.chalmers.dat265.group1.api.material;
+
+import org.apache.commons.lang3.NotImplementedException;
+import se.chalmers.dat265.group1.api.ApiController;
+import se.chalmers.dat265.group1.model.Material;
+import se.chalmers.dat265.group1.model.MaterialGrade;
+import se.chalmers.dat265.group1.model.dbEntities.MaterialData;
+import se.chalmers.dat265.group1.model.dbEntities.MaterialProperty;
+import se.chalmers.dat265.group1.storage.repository.GenericRepository;
+
+import java.util.*;
+
+public class MaterialsController extends ApiController implements MaterialAPI {
+    private GenericRepository<MaterialProperty> propertyRepository;
+
+
+    public MaterialsController(boolean debug) {
+        super(debug);
+        propertyRepository = new GenericRepository<>(MaterialProperty.class, dbConnector);
+    }
+
+    @Override
+    public List<Material> getAllMaterials() {
+        List<Material> materials = new LinkedList<>();
+
+        List<MaterialData> incompleteMaterials = materialRepository.getObjects();
+        for (MaterialData materialData : incompleteMaterials) {
+            materials.add(stageMaterial(materialData));
+        }
+
+        return materials;
+    }
+
+    private List<MaterialGrade> getMaterialGrades(int id) {
+        return materialGradeRepository.getObjects("materialID=" + id);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "materialID=" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)